semantics/cpp14/language/translation/name.k

```diff
@@ -497,6 +497,76 @@ module CPP-TRANSLATION-NAME
+
+     rule addBuiltinCandidates(operator<=, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 <= E2, BinaryOperator(operator<=, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator>=, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 >= E2, BinaryOperator(operator>=, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator==, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 == E2, BinaryOperator(operator==, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator!=, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 != E2, BinaryOperator(operator!=, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+*/
+
+
+     syntax Map ::= builtinNonvolatileEnumAssignOperator(Expr, CPPType, Expr, CPPType, ValueCategory) [function]
+     syntax Map ::= builtinVolatileEnumAssignOperator(Expr, CPPType, Expr, CPPType, ValueCategory) [function]
+     syntax Map ::= builtinNonvolatileArithAssignOperator(Expr, CPPType, ValueCategory, Expr, CPPType, ValueCategory) [function]
+
+     rule builtinNonvolatileEnumAssignOperator(E1::Expr, T1::CPPType, E2::Expr, T2::CPPType, VC::ValueCategory) =>
+                (type(functionType(type(simpleType(T1)),
+                      toCPPTypes(ListItem(type(lvRefType(type(simpleType(T1))))) ListItem(type(simpleType(T1)))),
+                      noMethod, CPPLinkage, exSet(.Set)))
+                |-> builtinOp(E1 := E2))
+
+          requires isCPPEnumType(T1) andBool notBool isVolatileType(T1) andBool viableType(type(simpleType(T1)), T2, VC)
+
+     rule builtinNonvolatileEnumAssignOperator(...) => .Map [owise]
+
+     rule builtinVolatileEnumAssignOperator(E1::Expr, T1::CPPType, E2::Expr, T2::CPPType, VC::ValueCategory) =>
+                (type(functionType(type(simpleType(T1)),
+                      toCPPTypes(ListItem(type(lvRefType(volatile(simpleType(T1))))) ListItem(type(simpleType(T1)))),
+                      noMethod, CPPLinkage, exSet(.Set)))
+                |-> builtinOp(E1 := E2))
+
+          requires isCPPEnumType(T1) andBool isVolatileType(T1) andBool viableType(type(simpleType(T1)), T2, VC)
+
+     rule builtinVolatileEnumAssignOperator(...) => .Map [owise]
+
+     rule builtinNonvolatileArithAssignOperator(E1::Expr, T1::CPPType, VC1::ValueCategory, E2::Expr, T2::CPPType, VC2::ValueCategory) =>
```

Comment:
this is wrong. as I said, in the case of arithmetic types, pretty much all integer types can be converted to from an enumeration type. Since the builtin operators are finite since there is a finite number of arithmetic types, you need to exhaustively enumerate the candidates as a constant of sort Map and add that constant to the mix and let overload resolution deal with it.